cmd/snap/cmd_run.go

```diff
@@ -67,7 +67,7 @@ func (x *cmdRun) Execute(args []string) error {
 	if x.Hook != "" && x.Command != "" {
 		return fmt.Errorf("cannot use --hook and --command together")
 	}
-	if x.Revision != "" && x.Hook == "" {
+	if x.Revision != "unset" && x.Revision != "" && x.Hook == "" {
```

Comment:
Haha, I don't know of valid use-cases for running hooks _with_ revisions! I've been assuming most of the time hooks would want to be run on the current revision. That assumption has been continuing into the `snap set` work, which so far only cares about the current revision.